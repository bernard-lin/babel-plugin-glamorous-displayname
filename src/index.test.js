import pluginTester from 'babel-plugin-tester'
import plugin from './'

pluginTester({
  plugin: plugin,
  snapshot: true,
  babelOptions: {filename: '/path/to/SomeComponent.js'},
  tests: {
    'does not affect non-named components': {
      code: `glamorous.button();`,
      snapshot: false,
    },
    'does not affect unused glamorous requires': {
      code: `const g = require('glamorous');`,
      snapshot: false,
    },
    'does not affect unused glamorous imports': {
      code: `
        import styled from 'styled-components';
        import glamorous from 'glamorous';
        glamorous.button();
        console.log(glamorous);
      `,
      snapshot: false,
    },
    'adds a display name for a component with a variable name': `
      import g from 'glamorous'
      const Button = g.button()
    `,
    'works with requires': `
      const l = require('glamorous')
      const MyDiv = l.div()
    `,
    'works with exported components': `
      import g from 'glamorous'
      export const MySection = g.section()
    `,
    'does not fail when destructuring built-in components': `
      import g from 'glamorous'
      const { Div } = g
    `,
    'inserts the displayName to the bottom of the closure': `
      import g from 'glamorous'

      function foo() {
        const MyDiv = g.div()
        console.log(MyDiv)
        return MyDiv
        // foo
      }
    `,
    'handles scope in try': `
      import g from 'glamorous'

      function foo() {
        try {
          const MyDiv = g.div()
          console.log(MyDiv)
        } catch(e) {
          // foo
        }
        return MyDiv
      }
    `,
    'handles try as the last function body item': `
      import g from 'glamorous'

      function foo() {
        const MyDiv = g.div()
        try {
          console.log(MyDiv)
          return MyDiv
        } catch(e) {
          // foo
        }
      }
    `,
    'handles finally': `
      import g from 'glamorous'

      function foo() {
        const MyDiv = g.div()
        try {
          console.log(MyDiv)
          return MyDiv
        } finally {
          console.log('other thing')
        }
      }
    `,
    'does not insert "unknown" as part of the displayName': {
      babelOptions: {filename: 'unknown'},
      code: `
        import g from 'glamorous'
        const MySection = g.section()
      `,
    },
  },
})
