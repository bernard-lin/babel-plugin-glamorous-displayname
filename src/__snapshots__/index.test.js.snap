// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`4. adds a display name for a component with a variable name 1`] = `
"
import g from 'glamorous'
const Button = g.button()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const Button = g.button();
Button.displayName = 'SomeComponent__Button';
"
`;

exports[`5. works with requires 1`] = `
"
const l = require('glamorous')
const MyDiv = l.div()

      ↓ ↓ ↓ ↓ ↓ ↓

const l = require('glamorous');
const MyDiv = l.div();
MyDiv.displayName = 'SomeComponent__MyDiv';
"
`;

exports[`6. works with exported components 1`] = `
"
import g from 'glamorous'
export const MySection = g.section()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
export const MySection = g.section();
MySection.displayName = 'SomeComponent__MySection';
"
`;

exports[`7. does not fail when destructuring built-in components 1`] = `
"
import g from 'glamorous'
const { Div } = g

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const { Div } = g;
"
`;

exports[`8. inserts the displayName to the bottom of the closure 1`] = `
"
import g from 'glamorous'

function foo() {
  const MyDiv = g.div()
  console.log(MyDiv)
  return MyDiv
  // foo
}

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';

function foo() {
  const MyDiv = g.div();
  console.log(MyDiv);
  MyDiv.displayName = 'SomeComponent__MyDiv';
  return MyDiv;
  // foo
}
"
`;

exports[`9. handles scope in try 1`] = `
"
import g from 'glamorous'

function foo() {
  try {
    const MyDiv = g.div()
    console.log(MyDiv)
  } catch(e) {
    // foo
  }
  return MyDiv
}

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';

function foo() {
  try {
    const MyDiv = g.div();
    console.log(MyDiv);
    MyDiv.displayName = 'SomeComponent__MyDiv';
  } catch (e) {
    // foo
  }
  return MyDiv;
}
"
`;

exports[`10. handles try as the last function body item 1`] = `
"
import g from 'glamorous'

function foo() {
  const MyDiv = g.div()
  try {
    console.log(MyDiv)
    return MyDiv
  } catch(e) {
    // foo
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';

function foo() {
  const MyDiv = g.div();
  try {
    console.log(MyDiv);
    MyDiv.displayName = 'SomeComponent__MyDiv';
    return MyDiv;
  } catch (e) {
    // foo
  }
}
"
`;

exports[`11. handles finally 1`] = `
"
import g from 'glamorous'

function foo() {
  const MyDiv = g.div()
  try {
    console.log(MyDiv)
    return MyDiv
  } finally {
    console.log('other thing')
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';

function foo() {
  const MyDiv = g.div();
  try {
    console.log(MyDiv);
    return MyDiv;
  } finally {
    console.log('other thing');
    MyDiv.displayName = 'SomeComponent__MyDiv';
  }
}
"
`;

exports[`12. does not insert "unknown" as part of the displayName 1`] = `
"
import g from 'glamorous'
const MySection = g.section()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const MySection = g.section();
MySection.displayName = 'MySection';
"
`;
